src/any_agent/serving/server.py

```diff
@@ -27,8 +38,35 @@ def _get_a2a_server(
     return A2AStarletteApplication(agent_card=agent_card, http_handler=request_handler)
 
 
-def serve_a2a(server: A2AStarletteApplication, host: str, port: int) -> None:
+def _create_server(
+    app: A2AStarletteApplication, host: str, port: int, log_level: str = "warning"
+) -> uvicorn.Server:
+    config = uvicorn.Config(app.build(), host=host, port=port, log_level=log_level)
+    return uvicorn.Server(config)
+
+
+async def serve_a2a_async(
+    server: A2AStarletteApplication, host: str, port: int, log_level: str = "warning"
+) -> tuple[asyncio.Task[Any], uvicorn.Server]:
+    """Provide an A2A server to be used in an event loop."""
+    uv_server = _create_server(server, host, port)
+    task = asyncio.create_task(uv_server.serve())
+    while not uv_server.started:  # noqa: ASYNC110
+        await asyncio.sleep(0.1)
+    return (task, uv_server)
+
+
+def serve_a2a(
+    server: A2AStarletteApplication, host: str, port: int, log_level: str = "warning"
+) -> None:
     """Serve the A2A server."""
-    import uvicorn
+    try:
+        loop = asyncio.get_running_loop()
+    except RuntimeError:
+        loop = asyncio.new_event_loop()
```

Comment:
To be consistent with how we use async in any_agent, can we do
```
return asyncio.get_event_loop().run_until_complete(
            self.run_async(prompt, **kwargs)
        )
```

If we want to change how we manage the asyncio loop, lets handle that in a separate PR, and change all the sync run/serve implementations.

Reply:
Ok, _now_ I get what you mean. I was protecting against the loop not having been created, but if you think that's not an issue I'll make the style consistent  👍🏼

Reply:
I'm not totally sure which design is better, but at least for this PR lets make the implementation consistent across any-agent, and we can consider a change to all implementations in any-agent at a later PR 🚀